### DISCUSSION TITLE
[Feature] Support for Google Takeout JSON Sidecars

### DISCUSSION THREAD

**Discussion by jjspierx** [immich-app/immich#5083] [Feature Request]
### The feature

Immich currently supports XMP sidecars to supplement exif data.  Many Immich users are migrating from Google Photos via Google Takeout which uses JSON sidecars.  It would be great if Immich supported using these sidecars.  Currently many of us are using scripts or writing small programs to extract this data and edit the exif of the corresponding image to add GPS coordinates, and  photo taken time, etc...   These JSON files also have a property indicating if the photo is  from partner sharing.  This could also be used to filter/show partner sharing photos from previous Google Photos settings.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jjspierx** [comment]
Here is an example JSON Sidecar:

It would be really cool if Immich had the option to use favorite, partnersharing, image views, photoTakenTime, and gps coordinates, description etc...


{
  "title": "IMG_2005.HEIC",
  "description": "A good description",
  "imageViews": "77",
  "photoTakenTime": {
    "timestamp": "1664140144",
    "formatted": "Sep 25, 2022, 9:09:04 PM UTC"
  },
  "geoData": {
    "latitude": 1.2345,
    "longitude": 5.4321,
    "altitude": 200.1234,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "geoDataExif": {
    "latitude": 1.2345,
    "longitude": 5.4321,
    "altitude": 200.1234,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "people": [{
    "name": "Mother"
  },
 {
    "name": "Daughter"
  }
],
  "favorited": true,
  "url": "https://photos.google.com/photo/asdf1234",
  "googlePhotosOrigin": {
    "fromPartnerSharing": {
    }
}

**Comment by dzg** [comment]
Is this gonna happen?
+1

**Comment by jamesmoey** [comment]
I am looking to implement this.

**Reply by bo0tzz** [reply]
@jamesmoey thanks for the interest! We're not really looking to implement this sort of support into Immich directly anymore. The immich-go tool currently does a very good job at handling takeouts, and on the long term we want to build native support for https://dtinit.org/ into Immich so that we can support more than only Google Takeout.
